api-reference/10 UI Widgets/GridBase/1 Configuration/onToolbarPreparing.md

```diff
@@ -99,7 +99,96 @@ The following code shows how you can customize the toolbar using this function.
     <dx-data-grid ...
         (onToolbarPreparing)="onToolbarPreparing($event)">
     </dx-data-grid>
-    
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxDataGrid ...
+            @toolbar-preparing="onToolbarPreparing"
+        />
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DxDataGrid, DxButton } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxButton
+        },
+        methods: {
+            onToolbarPreparing(e) {
+                let toolbarItems = e.toolbarOptions.items;
+                // Modifies an existing item
+                toolbarItems.forEach(function(item) {
+                    if (item.name === "saveButton") {
+                        // Change the item options here
+                    }
+                });
+
+                // Adds a new item
+                toolbarItems.push({
+                    widget: 'dxButton',
+                    options: {
+                        icon: 'user',
+                        onClick: function() {
+                            // ...
+                        }
+                    },
+                    location: 'after'
+                });
+            }
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DataGrid, Button } from 'devextreme-react/data-grid';
```

Comment:
Same note here.

Reply:
Recently, in a similar case in the DataGrid popup editing demo, we faced the issue that DxButton should be imported in React and Vue. So, I added it to code snippets.